library/firewall_lib.py

```diff
@@ -222,6 +224,48 @@
 except ImportError:
     HAS_FIREWALLD = False
 
+try:
+    from firewall.core.fw_nm import (
+        nm_is_imported,
+        nm_get_connection_of_interface,
+        nm_get_zone_of_connection,
+        nm_set_zone_of_connection,
+        nm_get_interfaces,
+        nm_get_client,
+    )
+
+    NM_IMPORTED = nm_is_imported()
+except ImportError:
+    NM_IMPORTED = False
+
+
+# Below: Adapted from firewall-cmd source code
+def try_set_zone_of_interface(module, _zone, interface):
+    if NM_IMPORTED:
+        try:
+            connection = nm_get_connection_of_interface(interface)
+        except Exception:
+            pass
+        else:
+            if connection is not None:
```

Comment:
Very nested. This should all probably be in its own function. With so many conditionals it can be hard to follow.

Reply:
I'll try to make it more readable